admin-console/src/main/webapp/src/components/EmailTemplates/InstantEmailTemplateModal.tsx

```diff
@@ -0,0 +1,82 @@
+import {
+    ActionGroup,
+    Button,
+    Form,
+    FormGroup,
+    FormSelect,
+    Modal,
+    ModalVariant
+} from '@patternfly/react-core';
+import React from 'react';
+
+import { InstantTemplate } from '../../types/Notifications';
+
+interface InstantTemplateModalProps {
+    showModal: boolean;
+    isEdit: boolean;
+    onClose: () => void;
+    templates: string[] | undefined;
+    initialInstantTemplate?: Partial<InstantTemplate>;
+    onSubmit: (instantTemplate: Partial<InstantTemplate>) => void;
+}
+
+export const InstantTemplateModal: React.FunctionComponent<InstantTemplateModalProps> = (props) => {
+
+    const [ instantTemplate, setInstantTemplate ] = React.useState<Partial<InstantTemplate>>(props.initialInstantTemplate ?? {});
+
+    const handleChange = (value: string, event: React.FormEvent<HTMLFormElement> | React.FormEvent<HTMLSelectElement>) => {
+        const target = event.target as HTMLSelectElement;
+        setInstantTemplate(prev => ({ ...prev, [target.name]: target.value }));
+    };
```

Comment:
This code is pretty generic, but could work nonetheless. Need to ensure that the values are being set on the form are what the instant template is expecting.